### PR TITLE
Changes needed to build with OpenSSL 1.1

### PR DESCRIPTION
**Pull request by hillu**
~~Please don't merge -- I am not sure about the `i2c_*` / `i2d_*` change yet.~~

### CODE REVIEW

**Comment by wxsBSD** [comment]
Thank you! I started down this path a few months ago but ended up getting sidetracked onto some other things and never got back to this!

**Comment by hillu** [comment]
@wxsBSD can you shed some light on what `i2c_ASN1_INTEGER()` is supposed to do here?

**Comment by wxsBSD** [comment]
It was used to convert the internal representation of the serial number to the human readable string form.

On a related note, there are a couple of places in that file where i2c_ is mentioned in the comments, you will want to clean those up too.

**Comment by hillu** [comment]
@wxsBSD I am not even sure if using `i2d_ASN1_INTEGER()` is the correct replacement -- do we want a hex dump of the DER-encoded INTEGER?

**Comment by wxsBSD** [comment]
It has been too long and I don't remember the terminology, so I can't say. We want the format that VirusTotal uses, which is also the format most other tolls use. I can double check this patch later today to verify it doesn't break some of my old signatures.

**Comment by hillu** [comment]
Apparently, `i2c_ASN1_INTEGER` puts the "raw data"  into a buffer while `i2d_ASN1_INTEGER` prefixes that data with DER type and length information. I have updated my patch to use `i2d_ASN1_INTEGER`. In my tests, I got the same result as before for `4bfe05f182aa273e113db6ed7dae4bb8`  (negative serial) and for a legit psexec binary.

This change can be merged, as far as I'm concerned.

File: libyara/modules/pe.c [hunk 1]
```diff
@@ -1336,6 +1344,10 @@ void pe_parse_certificates(
                     (char*) serial_ascii + 3 * j, 3, "%02x", serial_bytes[j]);
               }
 
+              fprintf(stderr, "signature[%i].serial = '%s'\n", counter, serial_ascii);
```
Comment (wxsBSD) [inline libyara/modules/pe.c]: Probably want to remove this. :)

**Comment by wxsBSD** [comment]
Thank you! I was going to look into this today (been backlogged at $job) but you seem to have covered my thoughts well. As long as serials are matching before and after this change I'm happy with it.

File: libyara/modules/pe.c (hunk 1 above)
Comment (hillu) [inline libyara/modules/pe.c]: done.

**Comment by wxsBSD** [comment]
I haven't tested on an OpenSSL 1.1 system (the VM has gone missing), but I did test on my OS X install running OpenSSL 0.9.8 (plus whatever patches Apple has applied) and tested it against a signed sample I had laying around. No regressions AFAICT. Thanks for looking into this, it would be great to get it into 3.5.0 as we will be seeing these OpenSSL changes become more widespread.

**Pull request merged**